File: src/openfl/events/ErrorEvent.nim
import ./Event
import ./TextEvent

type
  ErrorEvent* = ref object of TextEvent
    errorID*: int

proc newErrorEvent*(typeStr: string, bubbles: bool = false, cancelable: bool = false, text: string = "", id: int = 0): ErrorEvent =
  let self = ErrorEvent(errorID: id)
  self.`type` = typeStr
  self.bubbles = bubbles
  self.cancelable = cancelable
  self.text = text
  return self

method clone*(self: ErrorEvent): Event =
  let event = newErrorEvent(self.`type`, self.bubbles, self.cancelable, self.text, self.errorID)
  event.eventPhase = self.eventPhase
  event.target = self.target
  event.currentTarget = self.currentTarget
  return event

method toString*(self: ErrorEvent): string =
  "[ErrorEvent type=" & self.`type` & " bubbles=" & $self.bubbles & " cancelable=" & $self.cancelable & " text=" & self.text & " errorID=" & $self.errorID & "]"

const
  ERROR* = "error"

File: kfl/src/main/kotlin/openfl/events/ErrorEvent.kt
package openfl.events

open class ErrorEvent(
    type: String,
    bubbles: Boolean = false,
    cancelable: Boolean = false,
    text: String = "",
    val errorID: Int = 0
) : TextEvent(type, bubbles, cancelable, text) {

    companion object {
        const val ERROR = "error"
    }

    override fun clone(): ErrorEvent {
        val event = ErrorEvent(type, bubbles, cancelable, text, errorID)
        event.target = target
        event.currentTarget = currentTarget
        event.eventPhase = eventPhase
        return event
    }

    override fun toString(): String {
        return formatToString("ErrorEvent", "type", "bubbles", "cancelable", "text", "errorID")
    }
}
